frb_rust/src/macros.rs

```diff
@@ -0,0 +1,54 @@
+/// On WASM, [JsValue][wasm_bindgen::JsValue]s cannot be shared between workers but instead can be
+/// "transferred". Rust however is not aware of transferables and therefore cannot
+/// capture these values. This macro returns a [TransferClosure] on WASM platforms
+/// which will capture these special values, and a normal [FnOnce] on other platforms.
+/// Note that the parameter names must match available variables/bindings from the outer scope.
+#[macro_export]
+macro_rules! transfer {
```

Comment:
uber-nit: If I understand correctly, `transfer!` is used together w/ `TransferClosure` and its friends. However they are in different files currently. Shall we merge them together, such that readers may feel more natural.

Then, we can even make `TransferClosure` and its friends *non-pub*, since I see comments saying it is internal implementations

Reply:
`spawn!` now exists to replace `std::thread::spawn` and also used for internal implementations.